src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1529,6 +1530,30 @@ public function getContent($asResource = false)
         return $this->content;
     }
 
+    /**
+     * Return request body content as json decoded.
+     *
+     * @return mixed
+     *
+     * @throws JsonException
+     */
+    public function getJsonDecoded(int $depth = 512, int $options = 0)
+    {
+        $body = $this->getContent();
+
+        if ('' === $body) {
+            throw new JsonException('Unable to parse empty request body.');
+        }
+
+        $json = json_decode($body, true, $depth, $options);
+
+        if (JSON_ERROR_NONE !== $errorCode = json_last_error()) {
+            throw new JsonException(sprintf('Unable to parse JSON: %s.', json_last_error_msg()), $errorCode);
```

Comment:
Not sure we need the prefix - that's already a JsonException, and the message already contains "json" from test cases, isn't it?